x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/middleware.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Immutable } from '../../../../../common/endpoint/types';
+import { ImmutableMiddlewareAPI, ImmutableMiddlewareFactory } from '../../../../common/store';
+import { AppAction } from '../../../../common/store/actions';
+import {
+  AsyncResourceState,
+  StaleResourceState,
+  getLastLoadedResourceState,
+} from '../state/async_resource_state';
+import {
+  TrustedAppsListData,
+  TrustedAppsListPageState,
+} from '../state/trusted_apps_list_page_state';
+import { TrustedAppsHttpService, TrustedAppsService } from '../service';
+import { needsRefreshOfListData } from './selectors';
+import { TrustedAppsListResourceStateChanged } from './action';
+
+const createTrustedAppsListResourceStateChangedAction = (
+  newState: Immutable<AsyncResourceState<TrustedAppsListData>>
+): Immutable<TrustedAppsListResourceStateChanged> => ({
+  type: 'trustedAppsListResourceStateChanged',
+  payload: { newState },
+});
+
+const refreshList = async (
+  store: ImmutableMiddlewareAPI<TrustedAppsListPageState, AppAction>,
+  trustedAppsService: TrustedAppsService
+) => {
+  const list = store.getState().listView;
```

Comment:
you should use a selector to do this instead of accessing the store data directly - or use selectors below to get the data you need out of state.

Reply:
Fixed